src/pages/home/report/ReportActionsView.tsx

```diff
@@ -49,34 +51,72 @@ type ReportActionsViewProps = ReportActionsViewOnyxProps & {
 
     /** The report actions are loading newer data */
     isLoadingNewerReportActions?: boolean;
+
+    /** Whether the report is ready for comment linking */
+    isReadyForCommentLinking?: boolean;
 };
 
+const DIFF_BETWEEN_SCREEN_HEIGHT_AND_LIST = 120;
+const SPACER = 16;
+
 function ReportActionsView({
     report,
     session,
     parentReportAction,
-    reportActions = [],
+    reportActions: allReportActions = [],
     isLoadingInitialReportActions = false,
     isLoadingOlderReportActions = false,
     isLoadingNewerReportActions = false,
+    isReadyForCommentLinking = false,
 }: ReportActionsViewProps) {
     useCopySelectionHelper();
     const reactionListRef = useContext(ReactionListContext);
+    const route = useRoute<RouteProp<CentralPaneNavigatorParamList, typeof SCREENS.REPORT>>();
+    const reportActionID = route?.params?.reportActionID;
     const didLayout = useRef(false);
     const didSubscribeToReportTypingEvents = useRef(false);
-    const isFirstRender = useRef(true);
-    const hasCachedActions = useInitialValue(() => reportActions.length > 0);
-    const mostRecentIOUReportActionID = useMemo(() => ReportActionsUtils.getMostRecentIOURequestActionID(reportActions), [reportActions]);
+
     const network = useNetwork();
     const {isSmallScreenWidth} = useWindowDimensions();
+    const contentListHeight = useRef(0);
+    const layoutListHeight = useRef(0);
+    const {windowHeight} = useWindowDimensions();
+    const isFocused = useIsFocused();
     const prevNetworkRef = useRef(network);
     const prevAuthTokenType = usePrevious(session?.authTokenType);
-
+    const [isInitialLinkedView, setIsInitialLinkedView] = useState(!!reportActionID);
     const prevIsSmallScreenWidthRef = useRef(isSmallScreenWidth);
-
-    const isFocused = useIsFocused();
     const reportID = report.reportID;
-    const hasNewestReportAction = reportActions[0]?.isNewestReportAction;
+    const isLoading = (!!reportActionID && isLoadingInitialReportActions) || !isReadyForCommentLinking;
+
+    /**
+     * Retrieves the next set of report actions for the chat once we are nearing the end of what we are currently
+     * displaying.
+     */
+    const fetchNewerAction = useCallback(
+        (newestReportAction: OnyxTypes.ReportAction) => {
+            if (isLoadingNewerReportActions || isLoadingInitialReportActions) {
+                return;
+            }
+
+            Report.getNewerActions(reportID, newestReportAction.reportActionID);
+        },
+        [isLoadingNewerReportActions, isLoadingInitialReportActions, reportID],
+    );
+
+    const {
+        visibleReportActions: reportActions,
+        loadMoreReportActionsHandler,
```

Comment:
We don't need to be concerned with adding items above the linked message because this functionality is already performing seamlessly and is well virtualized. Hence, we're not batching items on fetchOlderActions. So, we just slice the report actions from the bottom up(from the oldest) to the linked message.